File: FoodApp/app/src/main/java/com/example/foodapp/MainActivity.java
package com.example.foodapp;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.util.Log;
import android.widget.*;
import android.os.Bundle;

import com.example.foodapp.databinding.ActivityMainBinding;

import java.sql.Connection;


public class MainActivity extends AppCompatActivity {
    protected ActivityMainBinding binding;
    static protected Connection connection;
    static protected String idUser = "";
    private ImageView imgSearch;
    private TextView txtSearch;
    private ImageView imgRecipes;
    private TextView txtRecipes;
    private ImageView imgCooking;
    private TextView txtCooking;
    @Override
    public void onBackPressed(){
        startActivity(new Intent(this, MainActivity.class));
        finish();
    }
    private void setOnClickViews(){
        binding.searchBtn.setOnClickListener(btn -> {
            goSearch(new SearchFragment());
        });

        binding.recipesBtn.setOnClickListener(btn -> {
            goRecipes(new RecipesFragment());
        });

        binding.cookingBtn.setOnClickListener(btn -> {
            goCooking(new CookingFragment());
        });
    }
    private void initViews(){
        imgSearch = findViewById(R.id.icon_search);
        txtSearch = findViewById(R.id.txt_search);
        imgRecipes = findViewById(R.id.icon_recipes);
        txtRecipes = findViewById(R.id.txt_recipes);
        imgCooking = findViewById(R.id.icon_cooking);
        txtCooking = findViewById(R.id.txt_cooking);

        if(idUser.equals("")) {
            Bundle bundle = getIntent().getExtras();
            idUser = bundle.getString("id");
        }
    }
    private void replaceFragment(Fragment frg){
        FragmentManager fm = getSupportFragmentManager();
        FragmentTransaction ft = fm.beginTransaction();
        ft.replace(R.id.frameMain, frg);
        ft.commit();
    }
    private void goSearch(Fragment frg){

        imgSearch.setImageDrawable(ContextCompat.getDrawable(getApplicationContext(),R.drawable.search_selected));
        txtSearch.setTextColor(ContextCompat.getColor(getApplicationContext(), R.color.orange));
        imgRecipes.setImageDrawable(ContextCompat.getDrawable(getApplicationContext(),R.drawable.recipe));
        txtRecipes.setTextColor(ContextCompat.getColor(getApplicationContext(), R.color.black));
        imgCooking.setImageDrawable(ContextCompat.getDrawable(getApplicationContext(),R.drawable.cooking));
        txtCooking.setTextColor(ContextCompat.getColor(getApplicationContext(), R.color.black));

        replaceFragment(frg);
    }
    private void goRecipes(Fragment frg){

        imgSearch.setImageDrawable(ContextCompat.getDrawable(getApplicationContext(),R.drawable.search));
        txtSearch.setTextColor(ContextCompat.getColor(getApplicationContext(), R.color.black));
        imgRecipes.setImageDrawable(ContextCompat.getDrawable(getApplicationContext(),R.drawable.recipe_selected));
        txtRecipes.setTextColor(ContextCompat.getColor(getApplicationContext(), R.color.orange));
        imgCooking.setImageDrawable(ContextCompat.getDrawable(getApplicationContext(),R.drawable.cooking));
        txtCooking.setTextColor(ContextCompat.getColor(getApplicationContext(), R.color.black));
        replaceFragment(frg);
    }
    private void goCooking(Fragment frg){

        imgSearch.setImageDrawable(ContextCompat.getDrawable(getApplicationContext(),R.drawable.search));
        txtSearch.setTextColor(ContextCompat.getColor(getApplicationContext(), R.color.black));
        imgRecipes.setImageDrawable(ContextCompat.getDrawable(getApplicationContext(),R.drawable.recipe));
        txtRecipes.setTextColor(ContextCompat.getColor(getApplicationContext(), R.color.black));
        imgCooking.setImageDrawable(ContextCompat.getDrawable(getApplicationContext(),R.drawable.cooking_selected));
        txtCooking.setTextColor(ContextCompat.getColor(getApplicationContext(), R.color.orange));
        replaceFragment(frg);
    }
    @Override
    protected void onDestroy(){
        try {
            connection.close();
        }
        catch(Exception e){
            Toast.makeText(this, e.getMessage(),Toast.LENGTH_LONG).show();
            Log.e("ERROR", e.getMessage());
        }
        super.onDestroy();
    }
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);

        // If there was previous data inserted into the create recipe form, we delete it upon
        // entering MainActivity
        SharedPreferences prefs = getApplicationContext().getSharedPreferences("recipeDetails", Context.MODE_PRIVATE);
        prefs.edit().clear().apply();

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        initViews();
        setOnClickViews();

            try {
                if(connection == null || connection.isClosed()) {
                    ConnectionDB con = new ConnectionDB();
                    connection = con.connect();
                }
            } catch (Exception e) {
                Log.e("ERROR", e.getMessage());
            }

        // APP STARTS ON THE RECIPES PAGE;
        replaceFragment(new RecipesFragment());

    }

}

File: FoodApp/app/src/main/java/com/example/foodapp/kotlin/MainActivity.kt
package com.example.foodapp.kotlin

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.fragment.app.Fragment
import com.example.foodapp.R
import com.example.foodapp.databinding.ActivityMainBinding
import java.sql.Connection

class MainActivity : AppCompatActivity() {
    private var binding: ActivityMainBinding? = null
    private lateinit var imgSearch: ImageView
    private lateinit var txtSearch: TextView
    private lateinit var imgRecipes: ImageView
    private lateinit var txtRecipes: TextView
    private lateinit var imgCooking: ImageView
    private lateinit var txtCooking: TextView
    override fun onBackPressed() {
        startActivity(Intent(this, MainActivity::class.java))
        finish()
    }

    private fun setOnClickViews() {
        binding!!.searchBtn.setOnClickListener { goSearch(SearchFragment()) }
        binding!!.recipesBtn.setOnClickListener { goRecipes(RecipesFragment()) }
        binding!!.cookingBtn.setOnClickListener { goCooking(CookingFragment()) }
    }

    private fun initViews() {
        imgSearch = findViewById(R.id.icon_search)
        txtSearch = findViewById(R.id.txt_search)
        imgRecipes = findViewById(R.id.icon_recipes)
        txtRecipes = findViewById(R.id.txt_recipes)
        imgCooking = findViewById(R.id.icon_cooking)
        txtCooking = findViewById(R.id.txt_cooking)
        if (idUser == "") {
            val bundle = intent.extras
            idUser = bundle!!.getString("id")
        }
    }

    private fun replaceFragment(frg: Fragment) {
        val fm = supportFragmentManager
        val ft = fm.beginTransaction()
        ft.replace(R.id.frameMain, frg)
        ft.commit()
    }

    private fun goSearch(frg: Fragment) {
        imgSearch.setImageDrawable(ContextCompat.getDrawable(applicationContext,
            R.drawable.search_selected
        ))
        txtSearch.setTextColor(ContextCompat.getColor(applicationContext, R.color.orange))
        imgRecipes.setImageDrawable(ContextCompat.getDrawable(applicationContext, R.drawable.recipe))
        txtRecipes.setTextColor(ContextCompat.getColor(applicationContext, R.color.black))
        imgCooking.setImageDrawable(ContextCompat.getDrawable(applicationContext,
            R.drawable.cooking
        ))
        txtCooking.setTextColor(ContextCompat.getColor(applicationContext, R.color.black))
        replaceFragment(frg)
    }

    private fun goRecipes(frg: Fragment) {
        imgSearch.setImageDrawable(ContextCompat.getDrawable(applicationContext, R.drawable.search))
        txtSearch.setTextColor(ContextCompat.getColor(applicationContext, R.color.black))
        imgRecipes.setImageDrawable(ContextCompat.getDrawable(applicationContext,
            R.drawable.recipe_selected
        ))
        txtRecipes.setTextColor(ContextCompat.getColor(applicationContext, R.color.orange))
        imgCooking.setImageDrawable(ContextCompat.getDrawable(applicationContext,
            R.drawable.cooking
        ))
        txtCooking.setTextColor(ContextCompat.getColor(applicationContext, R.color.black))
        replaceFragment(frg)
    }

    private fun goCooking(frg: Fragment) {
        imgSearch.setImageDrawable(ContextCompat.getDrawable(applicationContext, R.drawable.search))
        txtSearch.setTextColor(ContextCompat.getColor(applicationContext, R.color.black))
        imgRecipes.setImageDrawable(ContextCompat.getDrawable(applicationContext, R.drawable.recipe))
        txtRecipes.setTextColor(ContextCompat.getColor(applicationContext, R.color.black))
        imgCooking.setImageDrawable(ContextCompat.getDrawable(applicationContext,
            R.drawable.cooking_selected
        ))
        txtCooking.setTextColor(ContextCompat.getColor(applicationContext, R.color.orange))
        replaceFragment(frg)
    }

    override fun onDestroy() {
        try {
            connection!!.close()
        } catch (e: Exception) {
            Toast.makeText(this, e.message, Toast.LENGTH_LONG).show()
            Log.e("ERROR", e.message!!)
        }
        super.onDestroy()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // If there was previous data inserted into the create recipe form, we delete it upon
        // entering MainActivity
        val prefs = applicationContext.getSharedPreferences("recipeDetails", MODE_PRIVATE)
        prefs.edit().clear().apply()
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding!!.root)
        initViews()
        setOnClickViews()
        try {
            if (connection == null || connection!!.isClosed) {
                val con = ConnectionDB()
                connection = con.connect()
            }
        } catch (e: Exception) {
            Log.e("ERROR", e.message!!)
        }

        // APP STARTS ON THE RECIPES PAGE;
        replaceFragment(RecipesFragment())
    }

    companion object {
        var connection: Connection? = null
        var idUser: String? = ""
    }
}